lib/puppet/application/device.rb

```diff
@@ -169,7 +169,7 @@ def main
     end
     devices.each_value do |device|
       begin
-        Puppet.info "starting applying configuration to #{device.name} at #{device.url}"
+        Puppet.info "starting applying configuration to #{device.name}"
```

Comment:
If you care about losing the debug context and `device.uri` is a ruby URI, you could use `#scheme`, `#host`, etc to extract out the public bits. You could also create a new URI, and maybe clear the user part with `#userinfo=`?